internal/kafka/internal/services/kafka.go

```diff
@@ -1450,6 +1452,16 @@ func (k *kafkaService) ManagedKafkasRoutesTLSCertificate(kafkaRequest *dbapi.Kaf
 		return err
 	}
 
+	// check whether database update is needed
+	shouldUpdateCertInfoInTheDatabase = kafkaRequest.KafkasRoutesBaseDomainTLSKeyRef != certManagementOutput.TLSKeyRef ||
+		kafkaRequest.KafkasRoutesBaseDomainTLSCrtRef != certManagementOutput.TLSCertRef ||
+		shouldUpdateCertInfoInTheDatabase
```

Comment:
I'd prefer 

```suggestion
	shouldUpdateCertInfoInTheDatabase = shouldUpdateCertInfoInTheDatabase || 
		kafkaRequest.KafkasRoutesBaseDomainTLSKeyRef != certManagementOutput.TLSKeyRef ||
		kafkaRequest.KafkasRoutesBaseDomainTLSCrtRef != certManagementOutput.TLSCertRef ||
```

to avoid the string comparison if `shouldUpdateCertInfoInTheDatabase` is already `true`